### PR TITLE
feat(resolver): Add default Edition2024 to resolver v3

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

With #13776 done, we can now make MSRV-aware resolver the default for the new edition as part of #9930

### CODE REVIEW

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
I am forgetting. Do we have any previous discussion or FCP for making MSRV resolver the default in edition 2024?

File: tests/testsuite/rust_version.rs
Comment (weihanglo) [inline tests/testsuite/rust_version.rs:1]: Do we have a test for `workspace.resolver = "3"`, and should we have one?

Comment (epage) [reply tests/testsuite/rust_version.rs:1]: #13776 had tests

**Comment by epage** [comment]
> I am forgetting. Do we have any previous discussion or FCP for making MSRV resolver the default in edition 2024?

The RFC called for making it a default in the next edition.  My plan is to stabilize it before the window closes for 2024.  If not, we can adjust from there.

File: tests/testsuite/rust_version.rs
Comment (weihanglo) [reply tests/testsuite/rust_version.rs:1]: Yes, but not for `resolver = "3"` at workspace level specifically. Do we need that?

Comment (epage) [reply tests/testsuite/rust_version.rs:1]: So long as we have some tests for workspace/package resolver interactions, I don't think we need the tests for every resolver version.

Comment (weihanglo) [reply tests/testsuite/rust_version.rs:1]: Fair enough!

**Comment by weihanglo** [comment]
@bors r+

Thanks!

**Pull request merged**